pkg/agent/agent_test.go

```diff
@@ -708,3 +709,42 @@ func TestUpgradeRelease(t *testing.T) {
 		})
 	}
 }
+
+func TestNewConfigFlagsFromCluster(t *testing.T) {
+	testCases := []struct {
+		name   string
+		config rest.Config
+	}{
+		{
+			name: "bearer token remains for an https host",
+			config: rest.Config{
+				Host:        "https://example.com/",
+				APIPath:     "",
+				BearerToken: "foo",
+			},
+		},
+		{
+			name: "bearer token remains for an http host",
+			config: rest.Config{
+				Host:        "http://example.com/",
+				APIPath:     "",
+				BearerToken: "foo",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			restClientGetter := NewConfigFlagsFromCluster("namespace-a", &tc.config)
+
+			config, err := restClientGetter.ToRESTConfig()
+			if err != nil {
+				t.Fatalf("%+v", err)
+			}
+
+			if got, want := config.BearerToken, tc.config.BearerToken; got != want {
```

Comment:
at least I understand this test :sweat_smile:

Reply:
Heh, yeah - I was so happy to be able to reproduce the actual issue in a unit test before fixing it, once I'd found the issue.